src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -319,6 +319,35 @@ export class SearchView extends ViewPane {
 		}
 	}
 
+	private refreshInputs() {
+		this.pauseSearching = true;
+		this.searchWidget.setValue(this.viewModel.searchResult.query?.contentPattern.pattern ?? '');
+		this.searchWidget.setReplaceAllActionState(false);
+		this.searchWidget.toggleReplace(true);
+		this.inputPatternIncludes.setOnlySearchInOpenEditors(this.viewModel.searchResult.query?.onlyOpenEditors || false);
+		this.inputPatternExcludes.setUseExcludesAndIgnoreFiles(!this.viewModel.searchResult.query?.userDisabledExcludesAndIgnoreFiles || true);
+		this.searchIncludePattern.setValue('');
+		this.searchExcludePattern.setValue('');
+		this.pauseSearching = false;
+	}
+
+	public async importSearchResult(searchModel: SearchModel): Promise<void> {
+		// experimental: used by the quick access search to overwrite a search result
+		this.viewModel.searchResult = searchModel.searchResult;
+		searchModel.instantiateNewSearchResult();
```

Comment:
I'm not quite following this, you take the searchResult from the quick access' SearchModel but then that SearchModel creates a new SearchResult for itself?

Reply:
yes, since the picker will be closed after doing this, the model no longer needs the search result. The life cycle of a search result is a bit strange, in that a search result just clears and re-populates when a search happens instead of disposing and re-creating. So a search model seems to always make the assumption that a search result exists, and in this case when I'm de-referencing a search result to transfer to another search model, the original search model still needs one.

I agree that this is very strange. If I wanted to move a search result's data from one model to another, how would you expect this to happen? Since the original model doesn't need the results anymore, I was thinking that this process of literally moving the searchresult would be fine, although it seems sketchy. Alternatively, I could rebuild the `FolderMatch`, `FileMatch`, `Match`, etc. that was in the previous searchmodel- but I was thinking that it'd be more efficient to literally transfer it over? I'm open for suggestions, since I don't know what's best practice.

Reply:
I think I get it, so you are reusing the SearchResult object, then creating a new one on the search model to make sure that the result isn't referenced by two models.

>  Since the original model doesn't need the results anymore, I was thinking that this process of literally moving the searchresult would be fine, although it seems sketchy

If it can be really obvious that this is the case- like if you can transfer the SearchResult to the view immediately before disposing the search model, or something like that- then I think that would be fine. But if that's not the case then here are a couple ideas that I think would be easier to follow than this

- Can the whole SearchModel be used in the view instead of transferring SearchResult between models?
- add a `transferSearchResult` method to SearchModel, so that the details of how this works are kept inside the model, so that you don't have to know to call `instantiateNewSearchResult` from outside
- Add a `clone()` method to SearchResult and call that to get a copy and send that to the view

Reply:
> If it can be really obvious that this is the case- like if you can transfer the SearchResult to the view immediately before disposing the search model, or something like that

The thing with this is that the `SearchModel` that originally housed the `SearchResult` still exists after its result gets moved out. That's why I create a new searchresult. This model is reused when the pick is used again.

>Can the whole SearchModel be used in the view instead of transferring SearchResult between models?

I actually had this in an earlier version of this code, but I found it strange that the a whole SearchModel was being moved to be the SearchView's viewmodel. I think the code in `ISearchWorkbenchService` suggests that things that use it (such as the searchView) should follow a singleton approach, and I think that makes sense to me given the naming of `SearchModel`.

https://github.com/microsoft/vscode/blob/e7fa6c48afc5e2a585ec1290cff3deec35a873fc/src/vs/workbench/contrib/search/browser/searchModel.ts#L2310-L2316

>add a transferSearchResult method to SearchModel, so that the details of how this works are kept inside the model, so that you don't have to know to call instantiateNewSearchResult from outside

This could probably work. Like `public transferSearchResult(dest: SearchModel)` that would do the work here:
https://github.com/microsoft/vscode/blob/e7ad5651ac26fa00a40aa1e4010e81b92f655569/src/vs/workbench/contrib/search/browser/searchView.ts#L336-L337

>Add a clone() method to SearchResult and call that to get a copy and send that to the view

I also thought of this one, and my rationale behind not using it is because the SearchResult to be cloned would no longer be used anyways, so it'd be more performant to just move the original instead of clone it.